modules/orchestrator/services/addon/addon.go

```diff
@@ -1489,11 +1490,17 @@ func (a *Addon) Delete(userID, routingInstanceID string) error {
 			if addonSpec.Category != apistructs.AddonCustomCategory {
 				// 调用 scheduler API 删除
 				relationAddons, err := a.db.GetByOutSideInstanceID(routingInstance.RealInstance)
-
+				if err != nil {
+					logrus.Errorf("failed to GetByOutSideInstanceID, %+v", err)
+					return err
+				}
 				cInfo, err := a.bdl.GetCluster(routingInstance.Cluster)
 				if err != nil {
 					logrus.Errorf("get cluster info failed, cluster name: %s, error: %v", routingInstance.Cluster, err)
-					return err
+					//The addon can also be forcibly deleted, when the cluster is not exists
+					if err.(*errorresp.APIError).Code() != "NotFound" {
```

Comment:
do not hardcode. use struct in errorresp

Reply:
use errorresp.IsNotFound(err)